connectors/source.py

```diff
@@ -186,6 +186,15 @@ async def changed(self):
         """
         return True
 
+    async def validate_configuration(self):
+        """When called, validates configuration of the connector that is contained in self.configuration
+
+        If connector configuration is invalid, this method will raise an exception
+        with human-friendly and actionable description
+        """
+        # TODO: when validate_configuration is implemented everywhere, we should make this method raise NotImplementedError
+        pass
```

Comment:
Should this method do some basic validation, e.g. presence of mandatory fields, type match.

Reply:
This can also be moved here:
https://github.com/elastic/connectors-python/blob/fbceb616d446c314eb28a48130a3a836662bce2f/connectors/source.py#L118-L121

Reply:
I'm not sure about it - inheritance model here makes it easy to actually break the implementation. I'll need to do refactoring for this, which will make this PR bigger than it should.

What I mean - if validation logic is inside `BaseSource.validate_configuration`, then to do it connector class should call the base class implementation with `super.validate_configuration()`. It's quite easy not to do this and lose validation of types and such. So we'd have to probably break validation code into 2 methods:

```
class BaseSource
    async def validate_configutarion(self):
        # do generic validation for configuration 
        self.validate_source_specific_configuration() # then do validation of source-specific stuff.
```